explorer/interpreter/action.h

```diff
@@ -249,8 +254,9 @@ class TypeInstantiationAction : public Action {
 // local context.
 class WitnessAction : public Action {
  public:
-  explicit WitnessAction(Nonnull<const Witness*> witness)
-      : Action(Kind::WitnessAction), witness_(witness) {}
+  explicit WitnessAction(Nonnull<const Witness*> witness,
+                         std::optional<SourceLocation> source_loc)
```

Comment:
Can we make the `SourceLocation` mandatory for `WitnessAction` and for `InstantiateWitness`? Looking at the creations of `WitnessAction` below, it looks like we'll always have a source location.

Reply:
Yes, I made `SourceLocation` mandatory for `WitnessAction` and `InstantiateWitness` in one of the recent commits.
Thanks for pointing this out!